### PR TITLE
[META-51] Connect frontend to backend

### PR DESCRIPTION
**Pull request by cnk1dev**
### Description of issue
Updated overview and details pages to use real data provided by the backend instead of mock data.
Also did a lot of style fixes to make the UI match Figma as closely as possible.

### How I implemented
Wrote queries to fetch data, generated types and hooks for them using codegen and adapted the relevant components to match the new types.

### Other
- Added TruffleAI logo
- Removed example component + no longer used mock data
- Updated BASE_URL/ fixed environment variable
- Refactored some components + added comments for more clarity
- Added util functions for formatting dates & large numbers

### CODE REVIEW

File: src/components/page/details/ProjectInformation.tsx [hunk 1]
```diff
@@ -24,21 +30,26 @@ const ProjectInformation = ({ image, name, eli5, tags }: ProjectInformationProps
...
         iconColor="text-gray-100"
       />
     </div>
+
     <p className="text-12 font-light">{eli5}</p>
   </div>
 )
 
+ProjectInformation.defaultProps = {
+  image: undefined
```
Comment (pczern) [inline src/components/page/details/ProjectInformation.tsx:52]: ?

File: src/components/page/details/ProjectInformation.tsx [hunk 2]
```diff
@@ -2,19 +2,25 @@ import { StarIcon } from '@primer/octicons-react'
...
-// eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars -- TODO
+/**
+ * Top part of project detail page (logo, name, tags, bookmark button)
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
 const ProjectInformation = ({ image, name, eli5, tags }: ProjectInformationProps) => (
   <div className="border-b border-gray-800 px-7 py-4">
     <div className="mb-4 flex flex-row items-center justify-between">
       <div className="flex flex-row items-center">
+        {/* TODO: Replace with actual image */}
```
Comment (pczern) [inline src/components/page/details/ProjectInformation.tsx:19]: `@TODO`

File: src/components/page/details/ProjectInformation.tsx [hunk 3]
```diff
@@ -2,19 +2,25 @@ import { StarIcon } from '@primer/octicons-react'
 import Button from '../../pure/Button'
 
 type ProjectInformationProps = {
-  image: string // todo change type
+  image?: string
```
Comment (pczern) [inline src/components/page/details/ProjectInformation.tsx:5]: why optional

File: package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "dev": "next dev",
+    "dev": "next dev -p 3002",
```
Comment (pczern) [inline package.json:6]: change back to 3000

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/components/pure/Loading.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      <Image src={Logo} alt="logo" className="h-10 w-10" />
+      <p className="pt-6">{message || 'Loading...'}</p>
+    </div>
+
+    <div className="self-center pb-4 text-12 text-gray-300">© 2023 La Famiglia x Rostlab</div>
+  </div>
+)
+
+Loading.defaultProps = {
+  message: undefined
```
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/components/pure/Loading.tsx]: fixed

File: src/pages/details/[id].tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+ */
+const DetailPage: NextPage = () => {
+  // Get project id from URL
+  const {
+    query: { id }
+  } = useRouter()
+
+  return (
+    <Page>
+      <Details id={(typeof id === 'string' ? id : id?.join('')) || ''} />
```
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/pages/details/[id].tsx:18]: `const id: string | string[] | undefined`

File: src/pages/_app.tsx [hunk 7]
```diff
@@ -32,7 +32,12 @@ const App = ({
 export default withUrqlClient(
   () => ({
     url: BASE_URL,
-    exchanges: [cacheExchange, fetchExchange]
+    exchanges: [cacheExchange, fetchExchange],
+    fetchOptions: () => ({
+      headers: {
+        Authorization: document.cookie
```
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/pages/_app.tsx:38]: manually sends the supabase cookie as authorization header on every request made with the urql client

File: src/components/side-effects/LeftSidebar.tsx
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/components/side-effects/LeftSidebar.tsx:1]: renamed back to NavSidebar

File: src/components/pure/Error.tsx [hunk 8]
```diff
@@ -1,16 +1,27 @@
 type ErrorProps = {
-  title: string
-  message: string
+  title?: string
```
[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/components/pure/Error.tsx:2]: I added default props so you can just use the error component without supplying a title and message, it's likely gonna be the same standard message in most cases

File: src/components/page/details/ProjectInformation.tsx (hunk 1 above)
Comment (cnk1dev) [reply src/components/page/details/ProjectInformation.tsx]: fixed

File: src/components/page/details/ProjectInformation.tsx (hunk 2 above)
Comment (cnk1dev) [reply src/components/page/details/ProjectInformation.tsx]: fixed all TODOs

File: src/components/page/details/ProjectInformation.tsx (hunk 3 above)
Comment (cnk1dev) [reply src/components/page/details/ProjectInformation.tsx:5]: because some projects might not have an image_url, in that case we'll just display a placeholder image/ grey background

File: package.json (hunk 4 above)
Comment (cnk1dev) [reply package.json]: done

**Review by DamianGrosz2 — APPROVED** [review]

**Comment by DamianGrosz2** [comment]
Looks good

**Comment by cnk1dev** [comment]
Need to run codegen again before merging to update types but backend is down atm

File: src/components/pure/ProjectsTable/index.tsx
Comment (pczern) [inline src/components/pure/ProjectsTable/index.tsx:1]: Github has projects and repositories; is this a Repository Table or a Project Table?

File: src/pages/details/[id].tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (pczern) [reply src/pages/details/[id].tsx:18]: how can it be ever undefined?

[the remark this answers is not in the document]
Comment (cnk1dev) [reply src/pages/details/[id].tsx:18]: idk but it comes straight from `useRouter()` so what else should I do?

Screenshot 2023-05-30 at 12 36 24

File: src/components/pure/ProjectsTable/index.tsx
Comment (cnk1dev) [reply src/components/pure/ProjectsTable/index.tsx:1]: Imo we should use the term project and not repository in truffle, even if all projects are repositories atm. Project doesn't mean GitHub project though, but rather product or company. We might add other data sources in the future, so we should keep the terminology more general.

File: codegen.config.ts [hunk 9]
```diff
@@ -2,8 +2,9 @@ import type { CodegenConfig } from '@graphql-codegen/cli'
 import BASE_URL from './src/constants/baseUrl'
 
 const config: CodegenConfig = {
+  // @TODO environment variable doesn't work, fix
   schema: BASE_URL,
-  documents: './src/graphql/**/*.graphql',
+  documents: './src/queries/**/*.graphql',
```
Comment (pczern) [inline codegen.config.ts:7]: let's do `graphql/queries`

File: src/components/page/details/Chart.tsx [hunk 10]
```diff
@@ -37,89 +23,48 @@ const TimeframeOptions = [
...
-    return (
-      <div className="rounded border border-gray-800 bg-gray-900 p-2 text-xs text-white">
-        <p>{label}</p>
-        <p className="mt-2 font-bold">{payload[0].value}</p>
-      </div>
-    )
-  }
-
-  return null
+export type ChartProps = {
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:26]: I think an export is not needed as React has very good Utility functions to type components

File: src/components/page/details/Chart.tsx [hunk 11]
```diff
@@ -173,39 +118,52 @@ const Chart = ({ data }: ChartProps) => {
...
+              <Tooltip content={<CustomTooltip />} cursor={{ stroke: '#858699', strokeWidth: 1 }} />
 
               <Legend wrapperStyle={{ fontSize: '12px' }} />
 
-              <Line type="monotone" dataKey="value" stroke="#8884d8" activeDot={{ r: 8 }} />
-
-              {showSecondLine && <Line type="monotone" dataKey="value2" stroke="#82ca9d" />}
+              {chartData.map((dataset) => (
+                <Line
+                  key={dataset.name}
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:154]: i've already given up on `id`s

File: src/components/page/details/CustomTooltip.tsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { TooltipProps } from 'recharts'
+import formatNumber from '@/util/formatNumber'
+import formatDate from '@/util/formatDate'
+
+const CustomTooltip = ({ active, payload, label }: TooltipProps<string, string>) => {
+  if (active && payload && payload.length) {
+    return (
+      <div className="rounded border border-gray-800 bg-gray-900 p-2 text-xs text-white">
+        <p>{formatDate(label as string)}</p>
+        <p className="mt-2 font-bold">{formatNumber(parseFloat(payload[0].value as string))}</p>
```
Comment (pczern) [inline src/components/page/details/CustomTooltip.tsx:10]: this is a Tooltip component, how does it know about an array payload?
Design components without inherent design in mind. `body`,`title`

File: src/components/pure/Loading.tsx [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+type LoadingProps = {
+  message?: string
+}
+
+/**
+ * Simple loading component with logo and customizable message
+ */
+const Loading = ({ message }: LoadingProps) => (
+  <div className="flex h-screen grow flex-col items-center justify-between">
+    <div />
```
Comment (pczern) [inline src/components/pure/Loading.tsx:13]: what is this `<div />`

File: codegen.config.ts (hunk 9 above)
Comment (pczern) [reply codegen.config.ts:7]: and for codgen `graphql/generated`

Comment (cnk1dev) [reply codegen.config.ts:7]: done

File: src/components/page/details/Chart.tsx (hunk 10 above)
Comment (cnk1dev) [reply src/components/page/details/Chart.tsx:26]: used `React.ComponentProps` instead

File: src/components/pure/Loading.tsx (hunk 13 above)
Comment (cnk1dev) [reply src/components/pure/Loading.tsx:13]: for alignment

File: src/components/page/details/CustomTooltip.tsx (hunk 12 above)
Comment (cnk1dev) [reply src/components/page/details/CustomTooltip.tsx:10]: recharts requires a custom Tooltip component to be passed to its options, this component is only being used for that. I could wrap it in another component and pass down the props but don't see any advantage in doing that

File: src/components/page/details/Chart.tsx (hunk 11 above)
Comment (cnk1dev) [reply src/components/page/details/Chart.tsx]: added project id to dataset and used as key
